src/protocol/message.rs

```diff
@@ -179,3 +180,187 @@ fn checksum(bytes: &[u8]) -> u32 {
 
     u32::from_le_bytes(checksum)
 }
+
+/// Controls the filter response of [MessageFilter] to messages it reads.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Filter {
```

Comment:
Future consideration: we may want to move this code to another directory as it isn't explicitely part ot the protocol. Maybe `src/setup/message_filtering.rs` or similar? We'll also have some handshake helpers there and in future the synthetic nodes.

Reply:
Yes, I was quite uncertain about where to place it. Considered `src/setup/message_filtering.rs`, or creating a more utility focused module. Or even a module for generating fake messages which we may need in the future.

For now I just wanted to keep it compact while we figure it out organically.